### PR TITLE
[Fixes] Fixed script resolving local referenecs

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- The previous version of the script did not correctly resolve references in nested template files
- Regenerated all readmes

| Pipeline |
| - |
| [Compute - DiskEncryptionSets](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.diskencryptionsets.yml) |
| [Compute - VirtualMachines](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.virtualmachines.yml) |

# Type of Change

Please delete options that are not relevant.

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/compute/virtual-machines/README.md [hunk 1]
```diff
@@ -1026,6 +1026,8 @@ This section gives you an overview of all local-referenced module files (i.e., o
 
 | Reference | Type |
 | :-- | :-- |
```
Comment (AlexanderSehr) [inline modules/compute/virtual-machines/README.md:1028]: Blocking comment until singular names PR is merged

Comment (MariusStorhaug) [reply modules/compute/virtual-machines/README.md:1028]: I have merged your commits into the fail-fixing PR of mad proportions #3665

**Review by eriqua — DISMISSED** [review]
🆗

**Review by eriqua — COMMENTED** [review]
@AlexanderSehr all good except for conflict resolution

**Comment by AlexanderSehr** [comment]
Conflict resolved. Merging

**Pull request merged**